deployment/CONFIGURATION.md

```diff
@@ -79,6 +79,18 @@ Example config for installing only UI:
    `cd group_vars`
    2. Note the  <bridge_name> and add it to the hosts.yml configuration. For example, if a bridge file is named sokol-kovan.yml, you would change the <bridge_name> value in hosts.yml to sokol-kovan.
 
+## Monitor Configuration
+To deploy the monitor component there are two configuration steps required:
+1. Create the ansible configuration file `group_vars/<bridge_name>.yml` with the `MONITOR_PORT` variable.
+ 
+2. Create a configuration file `*.env` in `../monitor/configs` with the rest of the environmental variables detailed in the [monitor .env example](../monitor/.env.example). The monitor supports watching several bridges at the same time by creating one `.env` files for each bridge.
```

Comment:
I removed some of `ORACLE_*` vars that I incorrectly included in the examples for the monitor configs https://github.com/poanetwork/tokenbridge/pull/262/commits/b8b351943b9ce3a578585e9395736de6f7e4e4c9.
`ORACLE_*` variables are only needed once for the oracle configuration.

If we are going to have the oracle and monitor on the same system we need to specify `COMMON_*` variables twice. Once in `group_vars/<bridge_name>.yml` for the Oracle configuration and one in the `monitor/configs/<bridge_name>.env`. I'm not sure if this duplication can be reduced since there could be several bridges defined for the monitor so it means different definitions for the `COMMON_*` variable in each bridge `.env`

Reply:
What do you think if we for every bridge monitor should add its own:
```
---
<some-unique-name>:
  children:
    monitor:
      hosts:
        <host_ip>:
          ansible_user: ubuntu
``` 
in the `hosts.yml` file. All such elements will have the same IP address `<host_ip>`.

For example,
```
---
xdai:
  children:
    monitor:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
---
wetc:
  children:
    monitor:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
```

Then, for every element we will create `group_vars/<some-unique-name>.yml` with its own set of `COMMON_*` and `MONITOR_*` parameters (e.g. `group_vars/xdai.yml` and `group_vars/wetc.yml`)

By using the code allowing to deploy the monitor on the system where another monitor already exists it should be possible to add another instance of the monitor by one run of `ansible-playbook -i hosts.yml site.yml`. If I continue with the example above it will mean that from `ansible` point of view the deploy of the `wetc` monitor happens on a system where the `xdai` monitor is already deployed.

What do you think? Is it possible?

Reply:
I run some tests and I found one possible issue with the proposed approach.

Ansible Tasks are executed in order, one at a time, against all machines matched by the host pattern, before moving on to the next task. https://docs.ansible.com/ansible/latest/user_guide/playbooks_intro.html#tasks-list

Following the example, supposing that `155.24.155.24` is a clean host, when defining `hosts.yml` as:
```yaml
---
xdai:
  children:
    monitor:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
wetc:
  children:
    monitor:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
```

adding the proper `group_vars/xdai.yml` and `group_vars/wetc.yml` files, and running the playbooks with 
```bash
ansible-playbook -i hosts.yml site.yml
```
will have as a result that when running the task to detect if a monitor is already installed, the two hosts won't detect them as installed and will proceed to perform the complete actions for the deployment twice. This cause to later one of the hosts to fail in the step `Install apt dependencies` with the error message `Failed to lock apt for exclusive operation`.

The only way to make it work will be to run two times the ansible playbook (`ansible-playbook -i hosts.yml site.yml`). First with `xdai` configuration only in `hosts.yml` and later with `wetc` configuration only in `hosts.yml`.

Here is a comparison of the steps needed for deployment of oracle and 2 bridge monitors between the proposed approach and the current implemented in this PR code.

**Full Example of deployment of Oracle and 2 bridges monitor with proposed approach**
Create `group_vars/xdai.yml` with `COMMON_*`, `ORACLE_*` and `MONITOR_*` vars.

Set `hosts.yml` as:
```yaml
---
xdai:
  children:
    monitor:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
    oracle:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
```
Run playbooks for xdai: `ansible-playbook -i hosts.yml site.yml`

Create `group_vars/wetc.yml` with `COMMON_*` and `MONITOR_*` vars.
Set `hosts.yml` as:
```yaml
---
wetc:
  children:
    monitor:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
```
Run playbooks for wetc: `ansible-playbook -i hosts.yml site.yml`

This second run should be fast because the monitor is already deployed and it will only generate the `configs/wetc.env` for this new bridge.

**Full Example of deployment of Oracle and 2 bridges monitor with current implemented code**
Create `group_vars/xdai.yml` with `COMMON_*`, `ORACLE_*` and `MONITOR_PORT` vars.
Create `monitor/configs/xdai.env` with `COMMON_*` and `MONITOR_*` vars.
Create `monitor/configs/wetc.env` with `COMMON_*` and `MONITOR_*` vars.

Set `hosts.yml` as:
```yaml
---
xdai:
  children:
    monitor:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
    oracle:
      hosts:
        155.24.155.24:
          ansible_user: ubuntu
```
Run playbooks: `ansible-playbook -i hosts.yml site.yml`

In the current approach, we have only one playbook run but the mentioned `COMMON_*` vars repeated. While in the proposed approach vars are not repeated but we have to run the playbook twice. This second run should be fast because the monitor is already deployed and it will only generate the configs/wetc.env for this new bridge.

If you think it is OK running two times the playbooks vs repeating the env variables then I will proceed with the implementation of the changes

Reply:
Implemented the changes https://github.com/poanetwork/tokenbridge/pull/262/commits/cab446162a99d2774241b1acf81b2358d1a9ceac and updated the docs https://github.com/poanetwork/tokenbridge/pull/262/commits/d865fd35e40b90816492a805dbd782e979838f25